.github/workflows/ci.yaml

```diff
@@ -0,0 +1,46 @@
+name: CI Workflow
+
+on: [pull_request, workflow_dispatch]
+
+env:
+  GO_VERSION_BUILD: '1.20'
+  JAVA_VERSION: '11'
+
+jobs:
+  main:
+    name: CI
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        go-version: ['1.18', '1.19', '1.20']
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install SSH Key
+        uses: shimataro/ssh-key-action@v2
+        with:
+          key: ${{ secrets.SSH_PRIVATE_KEY }}
+          known_hosts: ${{ vars.SSH_KNOWN_HOSTS }}
+      - name: Install Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ env.JAVA_VERSION }}
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build
+        uses: ./.github/actions/build
+        with:
+          go-version: ${{ env.GO_VERSION_BUILD }}
+      - name: Generate SDK
+        uses: ./.github/actions/generate-sdk
+      - name: Install Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Lint
+        if: ${{ matrix.go-version == '1.19' || matrix.go-version == '1.20' }}
+        working-directory: ./sdk-repo-cloned
+        run: make lint skip-non-generated-files=true
```

Comment:
This was redundant as it was already done in the `generate-sdk` action.

We could even think about running `make test skip-non-generated-files=true` also insider the `generate-sdk` script and then its is also not needed here explicitly in the CI pipeline

Reply:
Currently `make lint` is modifying files, so it can be removed from `make generate-sdk`